internal/executor/helm/bin_runner.go

```diff
@@ -8,9 +8,19 @@ import (
 )
 
 func runHelmCLIBinary(ctx context.Context, cfg Config, args []string) (string, error) {
-	cmd := exec.CommandContext(ctx, helmBinaryName, args...)
-	cmd.Env = os.Environ()
-	cmd.Env = append(cmd.Env,
+	// Use full path if the PLUGIN_DEPENDENCY_DIR env variable is found.
+	// If not, Go will get $PATH for current process and try to look up the binary.
+	//
+	// Unfortunately, we cannot override PATH env variable for a go-plugin process as Hashicorp overrides envs during plugin startup.
```

Comment:
```suggestion
	// Unfortunately, we cannot override PATH env variable for a go-plugin process as Hashicorp overrides envs during plugin startup.
```

If you know the internal code then I can understand this comment, but without that it's strange that we cannot override env PATH as go-plugin overrides envs during startup, but somehow the `PLUGIN_DEPENDENCY_DIR` is available 🤔  😄 

would be good to rephrase it

Reply:
Right 😄 Will do!